snap/info.go

```diff
@@ -1167,6 +1168,13 @@ type HookInfo struct {
 	Explicit bool
 }
 
+type ComponentHookInfo struct {
```

Comment:
After reading more, it seems like it might be better to add information about components to the existing HookInfo struct.